src/Sarif/Autogenerated/AnnotatedCodeLocation.cs

```diff
@@ -36,7 +37,9 @@ public SarifNodeKind SarifNodeKind
         /// An identifier for the location, unique within the scope of the code flow within which it occurs.
         /// </summary>
         [DataMember(Name = "id", IsRequired = false, EmitDefaultValue = false)]
-        public string Id { get; set; }
+        [JsonConverter(typeof(Microsoft.CodeAnalysis.Sarif.Readers.AnnotatedCodeLocationIdConverter))]
+        [JsonProperty("id", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
```

Comment:
Here are the attributes that make it work, provided by multiple `AttributeHint`s.